### PR TITLE
Comment, Like, Follow AJAXify

### PR DESCRIPTION
**Pull request by YoungSong99**

### CODE REVIEW

File: app/controllers/comments_controller.rb
Comment (aizatibraimova) [inline app/controllers/comments_controller.rb:1]: Good job on ensuring your controllers respond to Javascript requests.

File: app/views/comments/destroy.js.erb [hunk 1]
```diff
@@ -0,0 +1,3 @@
+$("#<%= dom_id(@comment) %>").fadeOut(5000, function() {
```
Comment (SaraDawner2000) [inline app/views/comments/destroy.js.erb:1]: Looks good, but I think 5 seconds is a tad too slow. Maybe .5 seconds would be better?:)

File: app/views/comments/_form.html.erb
Comment (aizatibraimova) [inline app/views/comments/_form.html.erb:1]: Well done on ensuring `dom_id` helper method generates the id

File: Gemfile.lock
Comment (SaraDawner2000) [inline Gemfile.lock:1]: This is for your IDE, right? Not really important now, but throwing in a comment explaining this stuff would be useful for collaborators on projects!

File: app/controllers/follow_requests_controller.rb [hunk 2]
```diff
@@ -52,8 +53,9 @@ def update
   def destroy
     @follow_request.destroy
     respond_to do |format|
-      format.html { redirect_back fallback_location: root_url, notice: "Follow request was successfully destroyed." }
+      format.html { redirect_back fallback_location: root_url}
```
Comment (SaraDawner2000) [inline app/controllers/follow_requests_controller.rb:56]: Why did you remove the notice?

File: app/controllers/likes_controller.rb [hunk 3]
```diff
@@ -25,8 +25,9 @@ def create
 
     respond_to do |format|
       if @like.save
-        format.html { redirect_back fallback_location: root_url, notice: "Like was successfully created." }
+        format.html { redirect_back fallback_location: root_url}
```
Comment (SaraDawner2000) [inline app/controllers/likes_controller.rb:28]: same as above

File: app/controllers/likes_controller.rb [hunk 4]
```diff
@@ -51,8 +52,9 @@ def update
   def destroy
     @like.destroy
     respond_to do |format|
-      format.html { redirect_back fallback_location: root_url, notice: "Like was successfully destroyed." }
+      format.html { redirect_back fallback_location: root_url }
```
Comment (SaraDawner2000) [inline app/controllers/likes_controller.rb:55]: same as above

File: app/views/comments/edit.js.erb
Comment (aizatibraimova) [inline app/views/comments/edit.js.erb:1]: Great way of using helpers like `dom_id` and render to dynamically generate IDs and render HTML partials, integrating server-side logic with client-side JavaScript for a smooth editing experience.

**Review by aizatibraimova — APPROVED** [review]

File: app/views/comments/_comment.html.erb
Comment (SaraDawner2000) [inline app/views/comments/_comment.html.erb:1]: great job formatting code for readibility!

File: app/views/comments/_form.html.erb [hunk 5]
```diff
@@ -23,6 +23,9 @@
 
     <div class="actions">
       <%= form.submit class: "btn btn-outline-secondary btn-block" %>
+      <% if comment.persisted? %>
```
Comment (SaraDawner2000) [inline app/views/comments/_form.html.erb:26]: I'm interested in learning more about this!

File: app/views/comments/edit.js.erb
Comment (SaraDawner2000) [inline app/views/comments/edit.js.erb:1]: Great job going beyond basic requirements and adding the cancel button functionality!

File: app/views/comments/update.js.erb
Comment (SaraDawner2000) [inline app/views/comments/update.js.erb:1]: I appreciate the console log to track that everything performed well!

File: app/views/follow_requests/create.js.erb
Comment (SaraDawner2000) [inline app/views/follow_requests/create.js.erb:1]: It would be nice if you refactored this to improve readibility

File: app/views/follow_requests/destroy.js.erb
Comment (SaraDawner2000) [inline app/views/follow_requests/destroy.js.erb:1]: same as above

File: app/views/likes/create.js.erb
Comment (SaraDawner2000) [inline app/views/likes/create.js.erb:1]: same as above

File: app/views/likes/destroy.js.erb
Comment (SaraDawner2000) [inline app/views/likes/destroy.js.erb:1]: same as above

File: config/importmap.rb [hunk 6]
```diff
@@ -5,5 +5,5 @@
 pin "@hotwired/stimulus", to: "stimulus.min.js", preload: true
 pin "@hotwired/stimulus-loading", to: "stimulus-loading.js", preload: true
 pin_all_from "app/javascript/controllers", under: "controllers"
-pin "@rails/ujs", to: "https://ga.jspm.io/npm:@rails/ujs@7.0.8/lib/assets/compiled/rails-ujs.js"
+pin "@rails/ujs", to: "https://ga.jspm.io/npm:@rails/ujs@7.1.3-4/app/assets/javascripts/rails-ujs.esm.js"
```
Comment (SaraDawner2000) [inline config/importmap.rb:8]: is this also a change for your IDE?

**Review by SaraDawner2000 — APPROVED** [review]
Great job! Clean, easy to read code that shows dedication to UX beyond the basic requirements (the cancel button for editing comments)

File: app/controllers/follow_requests_controller.rb (hunk 2 above)
Comment (YoungSong99) [reply app/controllers/follow_requests_controller.rb:56]: Thanks for the comment! 
I realize that since our notice is placed at the top, most users don't see it. I'm considering switching to an alert or modal feature in the future. :)

File: Gemfile.lock
Comment (ahuynh3a) [reply Gemfile.lock:1]: I agree with Natalie! this addition was not necessary for the particular project, but adding a comment will help us understand why you added it.

File: app/controllers/comments_controller.rb [hunk 7]
```diff
@@ -59,23 +62,28 @@ def destroy
     respond_to do |format|
       format.html { redirect_back fallback_location: root_url, notice: "Comment was successfully destroyed." }
       format.json { head :no_content }
+
```
Comment (ahuynh3a) [inline app/controllers/comments_controller.rb:65]: Please remove any unnecessary white spaces to ensure clean code.

File: app/controllers/likes_controller.rb (hunk 4 above)
Comment (ahuynh3a) [reply app/controllers/likes_controller.rb:55]: In response to the notice changes, I say keep them as is because the changes were not necessary for this project!

File: app/views/comments/_form.html.erb [hunk 8]
```diff
@@ -1,5 +1,5 @@
-<li class="list-group-item">
-  <%= form_with(model: comment) do |form| %>
+<li id="<%= dom_id(comment.photo) %>_<%= dom_id(comment) %>_form" class="list-group-item">
+  <%= form_with(model: comment, local: false) do |form| %>
```
Comment (ahuynh3a) [inline app/views/comments/_form.html.erb:2]: great job for being mindful about when to use local:false and when to use remote:true

File: app/views/comments/create.js.erb [hunk 9]
```diff
@@ -0,0 +1,9 @@
+var added_comment = $("<%= j(render @comment) %>");
```
Comment (ahuynh3a) [inline app/views/comments/create.js.erb:1]: thank you for creating variables for your jquery!

File: app/views/comments/edit.js.erb
Comment (ahuynh3a) [inline app/views/comments/edit.js.erb:1]: I loved that you added an extra button to practice AJAX

File: app/views/comments/update.js.erb [hunk 10]
```diff
@@ -0,0 +1,2 @@
+$('#<%= dom_id(@comment.photo) %>_<%= dom_id(@comment) %>_form').replaceWith('<%= j render @comment %>');
```
Comment (ahuynh3a) [inline app/views/comments/update.js.erb:1]: This works! Consider putting this in a variable next time to make your code more legible.

File: app/views/follow_requests/create.js.erb [hunk 11]
```diff
@@ -0,0 +1 @@
+$(".follow_from_<%= dom_id(@follow_request.sender) %>_to_<%= dom_id(@follow_request.recipient) %>").replaceWith('<%= j(render "follow_requests/follow_unfollow", sender: @follow_request.sender, recipient: @follow_request.recipient) %>');
```
Comment (ahuynh3a) [inline app/views/follow_requests/create.js.erb:1]: Maybe refactor by putting in variables for readibility.

File: app/views/follow_requests/destroy.js.erb [hunk 12]
```diff
@@ -0,0 +1 @@
+$(".follow_from_<%= dom_id(@follow_request.sender) %>_to_<%= dom_id(@follow_request.recipient) %>").replaceWith('<%= j(render "follow_requests/follow_unfollow", sender: @follow_request.sender, recipient: @follow_request.recipient) %>');
```
Comment (ahuynh3a) [inline app/views/follow_requests/destroy.js.erb:1]: same as above

File: app/views/photos/_likes.html.erb [hunk 13]
```diff
@@ -5,7 +5,10 @@
     <% like = current_user.likes.find_by(photo: photo) %>
 
     <% if like %>
-      <%= link_to like, class: "btn btn-link link-underline-dark link-underline link-underline-opacity-0 link-underline-opacity-100-hover", data: { turbo_method: :delete } do %>
+      <%= link_to like,
```
Comment (ahuynh3a) [inline app/views/photos/_likes.html.erb:8]: Please remove any unnecessary white space

File: config/importmap.rb (hunk 6 above)
Comment (ahuynh3a) [reply config/importmap.rb:8]: I am a bit confused by this change, what is this for/

**Review by ahuynh3a — APPROVED** [review]
Hi Young, your code works great! I just added a few comments to keep in mind when you are coding. Nothing major! Good job overall!

File: Gemfile.lock
Comment (YoungSong99) [reply Gemfile.lock:1]: You are right!
.idea is used to prevent environment setup files and other project-specific files from being uploaded to Git. I added comment.

**Comment by YoungSong99** [comment]
@ahuynh3a @SaraDawner2000 @aizatibraimova 
Thanks for the review!
I reflect all the feedback and update the code. (https://github.com/YoungSong99/photogram-ajax/pull/4)
